cpp/core/jni/JniCommon.h

```diff
@@ -377,7 +377,13 @@ static inline void CheckException(JNIEnv* env) {
         env->GetStaticMethodID(describer_class, "describe", "(Ljava/lang/Throwable;)Ljava/lang/String;");
     std::string description =
         JStringToCString(env, (jstring)env->CallStaticObjectMethod(describer_class, describe_method, t));
-    throw gluten::GlutenException("Error during calling Java code from native code: " + description);
+    // Avoid C++ terminate when speculation kills a job.
+    // TODO: Find a better way rather than skip the exception.
+    if (description.find("ClosedByInterruptException") == std::string::npos) {
```

Comment:
Thanks for your fix!
Can we directly check whether `t` is `ClosedByInterruptException`? It may be more reliable.
It's ok if the current fix is not the final fix. Thanks!

Reply:
Good suggestion. I'm still working on it, and this is not a final fix. Please do not merge it temporarily.

Reply:
BTW, we cannot directly check `t` here, as there is other code that wraps an Exception on top of `t`, such as `LowCopyFileSegmentJniByteInputStream` would add `RuntimeException`.